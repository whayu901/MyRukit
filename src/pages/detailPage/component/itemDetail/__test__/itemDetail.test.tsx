import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import ItemDetail from "..";

jest.useFakeTimers();

// Note: test renderer must be required after react-native.

const mock = {
  name: "test",
  episode: ["hello", "world"],
  created: "sekarang bos",
};

describe("test itemDetail", () => {
  describe("rendering", () => {
    it("should match to snapshot - Text Transaction", () => {
      const itemDetail = renderer.create(
        <>
          <ItemDetail item={mock} indexItem={1} />
        </>,
      );
      expect({
        itemDetail,
      }).toMatchSnapshot("itemDetail snapshot true");
    });
  });
});
