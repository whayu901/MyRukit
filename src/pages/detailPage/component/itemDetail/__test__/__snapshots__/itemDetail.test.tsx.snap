// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test itemDetail rendering should match to snapshot - Text Transaction: itemDetail snapshot true 1`] = `
Object {
  "itemDetail": <View
    style={
      Object {
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Lekton-Bold",
          "fontSize": 16,
        }
      }
    >
      undefined - undefined
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Lekton-Regular",
          "fontSize": 12,
          "marginTop": 8,
        }
      }
    >
      August 20, 2017
    </Text>
  </View>,
}
`;
